GitUI/BranchTreePanel/RepoObjectsTree.Nodes.cs

```diff
@@ -319,13 +319,20 @@ private void FillTreeViewNode(string originalSelectedNodeFullNamePath, bool firs
                     var node = TreeViewNode.GetNodeFromPath(originalSelectedNodeFullNamePath);
                     if (node != null)
                     {
-                        if (((BaseBranchNode)node.Tag).Visible)
+                        if (node.Tag is BaseBranchNode branchNode)
                         {
-                            TreeViewNode.TreeView.SelectedNode = node;
+                            if (branchNode.Visible)
+                            {
+                                TreeViewNode.TreeView.SelectedNode = node;
+                            }
+                            else
+                            {
+                                TreeViewNode.TreeView.SelectedNode = null;
+                            }
                         }
                         else
                         {
-                            TreeViewNode.TreeView.SelectedNode = null;
+                            TreeViewNode.TreeView.SelectedNode = node;
                         }
```

Comment:
(is not with #8734)
```
                        TreeViewNode.TreeView.SelectedNode = !(node.Tag is BaseBranchNode branchNode) || branchNode.Visible
                            ? node
                            : null;
```

Reply:
Feels like I have to re-learn C# from scratch :)

Reply:
Updated